gooddata-flight-server/gooddata_flight_server/flexfun/flex_fun_execution_context.py

```diff
@@ -0,0 +1,105 @@
+# (C) 2024 GoodData Corporation
+from dataclasses import dataclass
+from typing import Optional
+
+from gooddata_sdk import Attribute, ComputeToSdkConverter, Filter, Metric
+
+
+@dataclass
+class ExecutionRequest:
+    """
+    Information about the execution request that is sent to the FlexFun.
+    """
+
+    attributes: list[Attribute]
+    """
+    All the attributes that are part of the execution request.
+    """
+
+    metrics: list[Metric]
+    """
+    All the metrics that are part of the execution request.
+    """
+
+    filters: list[Filter]
+    """
+    All the filters that are part of the execution request.
+    """
+
+    @staticmethod
+    def from_dict(d: dict) -> "ExecutionRequest":
+        """
+        Create ExecutionRequest from a dictionary.
+        :param d: the dictionary to parse
+        """
+        return ExecutionRequest(
+            attributes=[ComputeToSdkConverter.convert_attribute(a) for a in d.get("attributes", [])],
+            metrics=[ComputeToSdkConverter.convert_metric(m) for m in d.get("measures", [])],
+            filters=[ComputeToSdkConverter.convert_filter(f) for f in d.get("filters", [])],
+        )
+
+
+@dataclass
```

Comment:
Consider using attrs instead.

Reply:
Let's not add this dependency until we need some of its features